File: litcode/TwoSum/1회차.py
'''
https://leetcode.com/problems/two-sum/
숫자 배열과 타겟이 주어질 때
이 배열안의 2개 값으로 타겟이 되는 그 2개 값의 인덱스를 출력하라
'''
def f(nums, target):
  for i in  range(0, len(nums)):
    for j in range(i, len(nums)):
      if i == j:
        continue
      if (nums[i] + nums[j]) == target:
        return [i, j]

print(f([2,7,11,15], 9))
print(f([3,2,4], 6))
print(f([3,3], 6))
print(f([3,2,3], 6))

File: leetcode/2.AddTwoNumbers/1회차.js
/**
 * https://leetcode.com/problems/add-two-numbers/
 * 
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
 function ListNode(val, next) {
  this.val = (val === undefined ? 0 : val)
  this.next = (next === undefined ? null : next)
}

/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function (l1, l2) {
  let carry = 0
  let head = null
  let cur = null
  let next = null
  while(carry || l1 || l2) {
    const { val: val1 } = l1 || { val: 0 }
    const { val: val2 } = l2 || { val: 0 }
    let val = val1 + val2 + carry
    carry = Number.parseInt(val / 10)
    if (carry) {
      val -= 10
    }
    if (cur === null) {
      cur = new ListNode(val, null)
      head = cur
    } else {
      next = new ListNode(val, null)
      cur.next = next
      cur = next
    }

    l1 = l1 && l1.next ? l1.next : undefined
    l2 = l2 && l2.next ? l2.next : undefined
  } 
  return head
};

const l1 = new ListNode(9, new ListNode(9, new ListNode(9, null))) 
const l2 = new ListNode(9, new ListNode(9, new ListNode(9, null))) 
addTwoNumbers(l1, l2)